src/lib/validation/generic.ts

```diff
@@ -0,0 +1,81 @@
+// Only include methods that validate common entities and fields i.e. fields that are common to multiple forms across the service
+// These methods are then called by individual form validators that extend this class
+// Examples of fields common to multiple forms (to include in this class) are: email, username, phone number, postcode, gender, etc...
+
+import logger from "./../Logger";
+import errorManifest from "./../utils/error_manifests/default";
+
+export class GenericValidator {
+
+    errors: any;
+    payload: any;
+    errorManifest: any;
+
+    constructor () {
+        this.errors = this.getErrorSignature();
+        this.errorManifest = errorManifest;
+    }
+
+    protected getErrorSignature () {
+        return {
+            status: 400,
+            name: "VALIDATION_ERRORS",
+            message: errorManifest.validation.default.summary,
+            stack: {}
+        };
+    }
+
+    isValidEmail (email: string): boolean {
+        logger.info(`Request to validate email: ${email}`);
+        const regex = /^[-!#$%&'*+/0-9=?A-Z^_a-z{|}~](\.?[-!#$%&'*+/0-9=?A-Z^_a-z{|}~])*@[a-zA-Z](-?[a-zA-Z0-9])*(\.[a-zA-Z](-?[a-zA-Z0-9])*)+$/ig;
```

Comment:
So in ROE we originally had quite a complex reg exp like this (in both the web and API projects) but then ended up just having (in the web):

`export const VALID_EMAIL_FORMAT = /^.+@.+\..+$/;`

I think, from memory, because the more verbose one ended up being too strict. We went with the above as that is what the CH account repo uses (we also discussed it with @macreman ).